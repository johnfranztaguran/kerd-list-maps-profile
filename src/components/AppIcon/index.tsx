/**
 * app icon set component.
 * Usage: <AppIcon name="icon-name" size={20} color="#4F8EF7" />
 */
import React, { memo } from 'react'
import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import icoMoonConfig from './selection.json'

type Props = {
  name: string
  size: number
  color: string
}

const Icon = createIconSetFromIcoMoon(
  icoMoonConfig,
  'mdaid-icon',
  'mdaid-icon.ttf'
)

const AppIcon = memo(
  (props: Props): JSX.Element => {
    const { name, size = 20, color } = props
    return <Icon name={name} size={size} color={color} />
  }
)

export default AppIcon
