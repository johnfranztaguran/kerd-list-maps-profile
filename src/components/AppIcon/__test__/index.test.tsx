import React from 'react'
import { render } from '@testing-library/react-native'
import AppIcon from '../index'

describe('Component: AppIcon', () => {
  it('Should render a <AppIcon>', () => {
    const { baseElement } = render(
      <AppIcon name="ic-home" size={20} color="red" />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
