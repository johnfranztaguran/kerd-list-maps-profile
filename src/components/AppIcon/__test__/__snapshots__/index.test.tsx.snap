// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AppIcon Should render a <AppIcon> 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "mdaid-icon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Ó§ç
    </Text>
  </View>
</View>
`;
